account_invoice_validate_send_email/models/account_invoice.py

```diff
@@ -126,3 +137,11 @@ def _compute_carrier_tracking_refs(self):
                 for x in invoice.picking_ids
                 if x.carrier_tracking_ref
             )
+
+    @api.multi
+    def _compute_carrier_info(self):
+        for invoice in self:
+            picking = invoice.picking_ids[0]
+            if picking.carrier_info_id:
+                invoice.carrier_info_name = picking.carrier_info_id.name
+                invoice.carrier_info_url = picking.carrier_info_id.tracking_url
```

Comment:
```suggestion
        for invoice in self:
            carrier_recs = invoice.picking_ids.mapped("carrier_info_id")
            tracking_refs = invoice.picking_ids.mapped("carrier_tracking_ref")
            invoice.carrier_name = ", ".join(x.name for x in carrier_recs)
            invoice.carrier_tracking_url = ", ".join(x.tracking_url for x in carrier_recs)
            invoice.carrier_tracking_refs = ", ".join(tracking_refs)
```
Does this make sense (note that I haven't tested the code)? This should replace `_compute_carrier_tracking_refs()` as well.

Currently `_compute_carrier_tracking_refs()` takes care of the cases where an invoice involves multiple pickings. If that is the design we are after, we should also apply the same idea for the carrier name and the URL fields to take care of multiple pickings.

Reply:
Thank you, I added code and confirmed work.